File: library/src/test/java/io/github/anderscheow/validator/rules/common/NotEqualRuleTest.java
package io.github.anderscheow.validator.rules.common;

import android.support.annotation.StringRes;

import org.junit.After;
import org.junit.Before;
import org.junit.Test;

import java.util.Locale;

import static org.junit.Assert.*;

public class NotEqualRuleTest {

    private static final String KEYWORD = "test";

    private NotEqualRule notEqualRule;

    @Before
    public void setUp() throws Exception {
    }

    @After
    public void tearDown() throws Exception {
        notEqualRule = null;
    }

    @Test(expected = NullPointerException.class)
    public void validate_EmptySample_ThrowNullPointerException() throws Exception {
        notEqualRule = new NotEqualRule(KEYWORD);

        notEqualRule.validate(null);
    }

    @Test
    public void validate_ValidSample_ReturnTrue() throws Exception {
        notEqualRule = new NotEqualRule(KEYWORD);

        String sample = "TEST";

        assertTrue(notEqualRule.validate(sample));
    }

    @Test
    public void validate_InvalidSample_ReturnFalse() throws Exception {
        notEqualRule = new NotEqualRule(KEYWORD);

        String sample = "test";

        assertFalse(notEqualRule.validate(sample));
    }

    @Test
    public void errorMessage_DefaultErrorMessage() throws Exception {
        String errorMessage = String.format(Locale.getDefault(), "Value equal to '%s'", KEYWORD);

        notEqualRule = new NotEqualRule(KEYWORD);

        assertEquals(errorMessage, notEqualRule.errorMessage());
    }

    @Test
    public void errorMessage_CustomErrorMessage() throws Exception {
        String errorMessage = "This is a custom error message";

        notEqualRule = new NotEqualRule(KEYWORD, errorMessage);

        assertEquals(errorMessage, notEqualRule.errorMessage());
    }

    @Test
    public void errorMessage_CustomErrorRes() throws Exception {
        @StringRes int errorRes = 0;

        notEqualRule = new NotEqualRule(KEYWORD, errorRes);

        assertEquals(errorRes, notEqualRule.errorRes());
    }
}

File: library/src/test/java/io/github/anderscheow/validator/rules/common/NotEqualRuleTest.kt
package io.github.anderscheow.validator.rules.common

import androidx.annotation.StringRes
import org.junit.After
import org.junit.Assert.*
import org.junit.Before
import org.junit.Test

class NotEqualRuleTest {

    private lateinit var notEqualRule: NotEqualRule

    @Before
    @Throws(Exception::class)
    fun setUp() {
    }

    @After
    @Throws(Exception::class)
    fun tearDown() {
    }

    @Test(expected = NullPointerException::class)
    @Throws(Exception::class)
    fun validate_EmptySample_ThrowNullPointerException() {
        notEqualRule = NotEqualRule(INT_KEYWORD, "")

        notEqualRule.validate(null)
    }

    @Test
    @Throws(Exception::class)
    fun validate_StringSample_ReturnTrue() {
        notEqualRule = NotEqualRule(STRING_KEYWORD, "")

        val sample = "TEST"

        assertTrue(notEqualRule.validate(sample))
    }

    @Test
    @Throws(Exception::class)
    fun validate_StringSample_ReturnFalse() {
        notEqualRule = NotEqualRule(STRING_KEYWORD, "")

        val sample = "test"

        assertFalse(notEqualRule.validate(sample))
    }

    @Test
    @Throws(Exception::class)
    fun errorMessage_CustomErrorMessage() {
        val errorMessage = "This is a custom error message"

        notEqualRule = NotEqualRule(STRING_KEYWORD, errorMessage)

        assertEquals(errorMessage, notEqualRule.errorString)
    }

    @Test
    @Throws(Exception::class)
    fun errorMessage_CustomErrorRes() {
        @StringRes val errorRes = 0

        notEqualRule = NotEqualRule(STRING_KEYWORD, errorRes)

        assertEquals(errorRes, notEqualRule.errorRes)
    }

    companion object {
        private const val STRING_KEYWORD = "test"
        private const val INT_KEYWORD = 456
    }
}